pkg/bmc/ilo4.go

```diff
@@ -0,0 +1,93 @@
+package bmc
+
+import (
+	"net/url"
+	"strings"
+)
+
+func init() {
+	registerFactory("ilo4", newILOAccessDetails, []string{"http", "https"})
```

Comment:
It doesn't appear to me that the Ironic driver actually offers plain HTTP as an option.

Reply:
iLO driver does support HTTP as an option.

Reply:
@zaneb  You were right ilo does not support HTTP as an option. So removed it now.